test/channels.jl

```diff
@@ -76,6 +76,13 @@ end
     @test isopen(c)
     @test collect(c) == 1:100
 end
+@testset "Multithreaded task constructors" begin
+    taskref = Ref{Task}()
+    c = Channel(spawn=true, taskref=taskref) do c; put!(c, 0); end
+    # Test that the task is using the multithreaded scheduler
+    @test taskref[].sticky == false
+    @test collect(c) == [0]
+end
```

Comment:
I don't love that this test is exposing an implementation detail (tests like this make it more annoying to change implementation details like that), but I don't know of a better way to test that the task is created using the multithreaded scheduler?